mockwebserver/src/main/kotlin/mockwebserver3/MockWebServer.kt

```diff
@@ -849,6 +856,23 @@ class MockWebServer : Closeable {
 
     writeHeaders(sink, response.headers)
 
+    if (response.socketHandler != null) {
+      response.socketHandler.handle(
+        object : okio.Socket {
+          override val source: BufferedSource
```

Comment:
I'm not clear who can access this as a BufferedSource for an anonymous object impl. Wouldn't clients still have to Buffer again to be sure?

Reply:
I guess I don't understand your question. Are you implying that this should be changed to a non-buffered Source?